### PR TITLE
add custom kernel and infrastructure for raspberry pi 4 configuration

### PR DESCRIPTION
**Pull request by cnorthway**
configured with i2c, eudev, and serial

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,7 +74,12 @@ make --directory=$(pwd)/buildroot O=$(pwd)/${DEFCONFIG}_output savedefconfig # s
 ```
 
 ### kernel
-TODO
+```sh
+. env.sh
+make --directory=$(pwd)/buildroot O=$(pwd)/${DEFCONFIG}_output linux-menuconfig
+make --directory=$(pwd)/buildroot O=$(pwd)/${DEFCONFIG}_output linux-savedefconfig
+make --directory=$(pwd)/buildroot O=$(pwd)/${DEFCONFIG}_output linux-update-defconfig
```
Comment (jleightcap) [inline README.md:81]: huh weird this has 2 steps (`*-savedefconfig` and `*-update-defconfig`) vs. the buildroot single `savedefconfig`.

File: accessibility-oscilloscope/configs/rpi4_defconfig [hunk 2]
```diff
@@ -10,7 +12,8 @@ BR2_ROOTFS_POST_SCRIPT_ARGS="--add-miniuart-bt-overlay"
 BR2_LINUX_KERNEL=y
 BR2_LINUX_KERNEL_CUSTOM_TARBALL=y
 BR2_LINUX_KERNEL_CUSTOM_TARBALL_LOCATION="$(call github,raspberrypi,linux,967d45b29ca2902f031b867809d72e3b3d623e7a)/linux-967d45b29ca2902f031b867809d72e3b3d623e7a.tar.gz"
-BR2_LINUX_KERNEL_DEFCONFIG="bcm2711"
+BR2_LINUX_KERNEL_USE_CUSTOM_CONFIG=y
+BR2_LINUX_KERNEL_CUSTOM_CONFIG_FILE="../accessibility-oscilloscope/configs/rpi4_linux_defconfig"
```
Comment (jleightcap) [inline accessibility-oscilloscope/configs/rpi4_defconfig:16]: is this kernel config based on the stock  `bcm2711` kconfig then? as in, we only add stuff, and this won't regress anything?

**Review by jleightcap — APPROVED** [review]
nice!

File: README.md (hunk 1 above)
Comment (jleightcap) [reply README.md:81]: makes me wonder if we're missing a step for the regular buildroot config.

Comment (cnorthway) [reply README.md:81]: very strange yes, one goes from `.config` -> `defconfig` in the linux build dir, the other copies it to the buildroot custom location

File: accessibility-oscilloscope/configs/rpi4_defconfig (hunk 2 above)
Comment (cnorthway) [reply accessibility-oscilloscope/configs/rpi4_defconfig:16]: yes it's based on what I had locally modified before, should have been just our changes on top of `bcm2711`.
bit of bloat possibly with all the modules & things we don't need but I didn't want to touch it

File: README.md (hunk 1 above)
Comment (jleightcap) [reply README.md:81]: ahh ok there's two levels of redirection. nice, this seems great for version controlling kconfigs.

**Pull request merged**